src/net/sourceforge/kolmafia/persistence/ConcoctionDatabase.java

```diff
@@ -167,6 +168,38 @@ public class ConcoctionDatabase {
 
   private ConcoctionDatabase() {}
 
+  public static void registerShops() {
+    // Mixing methods that use shop.php
+    ShopDatabase.registerShop("5dprinter", "Xiblaxian 5D printer", CraftingType.FIVE_D);
```

Comment:
is the intent to move these into data files?

Reply:
Yes. My next PR will be refactoring shops.txt and shoprows.txt.
I needed this method to generate the lines for shops.txt - and also for setting up shopId <-> crafting type maps.
I'll get rid of this at that time.